salt/fileclient.py

```diff
@@ -1283,6 +1314,25 @@ def __hash_and_stat_file(self, path, saltenv='base'):
                 'cmd': '_file_hash_and_stat'}
         return self.channel.send(load)
 
+    def stat_file(self, path, saltenv='base'):
+        '''
+        '''
+        try:
+            path = self._check_proto(path)
+        except MinionError as err:
+            if not os.path.isfile(path):
+                msg = 'specified file {0} is not present to check stats: {1}'
+                log.warning(msg.format(path, err))
+                return []
+            else:
+                return list(os.stat(path))
+
+        if saltenv in self.remote_file_list and path in self.remote_file_list[saltenv]:
+            log.debug("*** Found cached stat! for file: " + path)
```

Comment:
Yeah the exclamation point has no purpose that's just the way I like to debug stuff :P I'll correct that